### PR TITLE
Improve GUC handling

### PR DESCRIPTION
**Pull request by cevian**
This PR improves GUC handling to be more inline with the Postgres GUC
system.

### CODE REVIEW

File: src/guc.c [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+							 NULL,
+							 &guc_disable_optimizations,
+							 false,
+							 PGC_USERSET,
+							 0,
+							 NULL,
+							 NULL,
+							 NULL);
+	DefineCustomBoolVariable("timescaledb.optimize_plain_tables",
+						"Apply timescale query optimization to plain tables",
```
Comment (olofr) [inline src/guc.c]: I know we use plain table internally but I wonder if it is intuitive for others ?

Comment (cevian) [inline src/guc.c]: Do you have a naming suggestion?

Comment (olofr) [inline src/guc.c]: Not a good one :) Non-hypertable maybe ? Then we at least don't introduce a new term.

Comment (olofr) [inline src/guc.c]: Actually, maybe non-hypertable is the correct term as it is the only thing we can actually say about tables outside of our control ?

Comment (RobAtticus) [inline src/guc.c]: `optimize_non_hypertables` seems good to me

Comment (erimatnor) [inline src/guc.c]: +1 for `optimize_non_hypertables`

Comment (cevian) [inline src/guc.c]: Fixed

**Pull request merged**